File: VirtualWorld-Cpp/Virtual-World/Antylopa.cpp
#include "Owca.h"
#include "Zwierze.h"
#include <iostream>
#include "Organizm.h"
#include "Antylopa.h"

Antylopa::Antylopa()
{
}

Antylopa::Antylopa(Swiat* swiat, std::vector<unsigned int> poz, int wiek, int inicj, int moc)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeWiek(wiek);
	changeInicjatywa(inicj);

	getSwiat()->edytujPlansze(poz[0], poz[1], ANTYLOPA);
}

Antylopa::Antylopa(Swiat* swiat, std::vector<unsigned int> poz)
{
	changeSwiat(swiat);
	changePolozenie(poz[0], poz[1]);
	changeInicjatywa(4);
	zmienSile(4);
	changeWiek(0);

	getSwiat()->edytujPlansze(poz[0], poz[1], ANTYLOPA);
}

void Antylopa::rysowanie()
{
	std::cout << ANTYLOPA;
}

char Antylopa::getZnak()
{
	return ANTYLOPA;
}

int Antylopa::zasieg()
{
	return 2;
}

bool Antylopa::ucieczka(std::vector<int>& teoretyczna, std::vector <std::vector <Organizm*>>& organizmy)
{

	int temp = rand() % 2;

	if (temp) {
		getSwiat()->addToLog("Jedna z antylop uciekla przed walka!");

		if (teoretyczna[1] - 1 >= 0 && getSwiat()->sprawdzPole(teoretyczna[0], teoretyczna[1] - 1) == ' ') {
			getSwiat()->edytujPlansze(getPolozenie(0), getPolozenie(1), ' ');
			getSwiat()->edytujPlansze(teoretyczna[0], teoretyczna[1] - 1, this->getZnak());
			getSwiat()->changePositionOrganizm(getPolozenie(0), getPolozenie(1), teoretyczna[0], teoretyczna[1] - 1, organizmy);
		}

		else if (teoretyczna[1] + 1 < getSwiat()->getRozmiarY() && getSwiat()->sprawdzPole(teoretyczna[0], teoretyczna[1] + 1) == ' ') {
			getSwiat()->edytujPlansze(getPolozenie(0), getPolozenie(1), ' ');
			getSwiat()->edytujPlansze(teoretyczna[0], teoretyczna[1] + 1, this->getZnak());
			getSwiat()->changePositionOrganizm(getPolozenie(0), getPolozenie(1), teoretyczna[0], teoretyczna[1] + 1, organizmy);
		}

		else if (teoretyczna[0] - 1 >= 0 && getSwiat()->sprawdzPole(teoretyczna[0] - 1, teoretyczna[1]) == ' ') {
			getSwiat()->edytujPlansze(getPolozenie(0), getPolozenie(1), ' ');
			getSwiat()->edytujPlansze(teoretyczna[0] - 1, teoretyczna[1], this->getZnak());
			getSwiat()->changePositionOrganizm(getPolozenie(0), getPolozenie(1), teoretyczna[0] - 1, teoretyczna[1], organizmy);
		}

		else if (teoretyczna[0] + 1 < getSwiat()->getRozmiarX() && getSwiat()->sprawdzPole(teoretyczna[0] + 1, teoretyczna[1] - 1) == ' ') {
			getSwiat()->edytujPlansze(getPolozenie(0), getPolozenie(1), ' ');
			getSwiat()->edytujPlansze(teoretyczna[0] + 1, teoretyczna[1], this->getZnak());
			getSwiat()->changePositionOrganizm(getPolozenie(0) + 1, getPolozenie(1), teoretyczna[0], teoretyczna[1], organizmy);
		}

		return 1;

	}

	return 0;

}

Antylopa::~Antylopa()
{
}

File: VirtualWorld-Java/src/Antylopa.java
import java.util.ArrayList;
import java.util.Random;

public class Antylopa extends Zwierze{
	
	public Antylopa() {
	}
		
		public Antylopa(Swiat swiat, ArrayList<Integer> poz, int wiek, int inicj, int moc) {
			changeSwiat(swiat);
			changePolozenie(poz.get(0), poz.get(1));
			changeWiek(wiek);
			changeInicjatywa(inicj);
			
			String s = ""+'A';
			getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
		}
		
		public Antylopa(Swiat swiat, ArrayList<Integer> poz) {
			changeSwiat(swiat);
			changePolozenie(poz.get(0), poz.get(1));
			changeInicjatywa(4);
			zmienSile(4);
			changeWiek(0);
			
			String s = ""+'A';
			getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
		}
		
		@Override
		public void rysowanie() {
			System.out.println('A');
		}
		
		@Override
		public char getZnak() {
			return 'A';
		}
		
		@Override
		public int zasieg() {
			return 2;
		}
		
		@Override
		public Boolean ucieczka(ArrayList<Integer> teoretyczna, ArrayList<ArrayList<Organizm>> organizmy) {
			Random rand = new Random();
			int temp = rand.nextInt(2);
			
			String s = ""+this.getZnak();
			
			if (temp >= 1) {
				getSwiat().addToLog("Jedna z antylop uciekla przed walka!");

				if (teoretyczna.get(1) - 1 >= 0 && getSwiat().sprawdzPole(teoretyczna.get(0), teoretyczna.get(1) - 1) == ' ') {
					getSwiat().edytujPlansze(getPolozenie(0), getPolozenie(1), " ");
					getSwiat().edytujPlansze(teoretyczna.get(0), teoretyczna.get(1) - 1, s);
					getSwiat().changePositionOrganizm(getPolozenie(0), getPolozenie(1), teoretyczna.get(0), teoretyczna.get(1) - 1, organizmy);
				}

				else if (teoretyczna.get(1) + 1 < getSwiat().getRozmiarY() && getSwiat().sprawdzPole(teoretyczna.get(0), teoretyczna.get(1) + 1) == ' ') {
					getSwiat().edytujPlansze(getPolozenie(0), getPolozenie(1), " ");
					getSwiat().edytujPlansze(teoretyczna.get(0), teoretyczna.get(1) + 1, s);
					getSwiat().changePositionOrganizm(getPolozenie(0), getPolozenie(1), teoretyczna.get(0), teoretyczna.get(1) + 1, organizmy);
				}

				else if (teoretyczna.get(0) - 1 >= 0 && getSwiat().sprawdzPole(teoretyczna.get(0) - 1, teoretyczna.get(1)) == ' ') {
					getSwiat().edytujPlansze(getPolozenie(0), getPolozenie(1), " ");
					getSwiat().edytujPlansze(teoretyczna.get(0) - 1, teoretyczna.get(1), s);
					getSwiat().changePositionOrganizm(getPolozenie(0), getPolozenie(1), teoretyczna.get(0) - 1, teoretyczna.get(1), organizmy);
				}

				else if (teoretyczna.get(0) + 1 < getSwiat().getRozmiarX() && getSwiat().sprawdzPole(teoretyczna.get(0) + 1, teoretyczna.get(1) - 1) == ' ') {
					getSwiat().edytujPlansze(getPolozenie(0), getPolozenie(1), " ");
					getSwiat().edytujPlansze(teoretyczna.get(0) + 1, teoretyczna.get(1), s);
					getSwiat().changePositionOrganizm(getPolozenie(0) + 1, getPolozenie(1), teoretyczna.get(0), teoretyczna.get(1), organizmy);
				}

				return true;

			}

			return false;
		}
	
}
